src/excmds.ts

```diff
@@ -3072,7 +3072,38 @@ export async function bmark(url?: string, ...titlearr: string[]) {
     dupbmarks.map(bookmark => browser.bookmarks.remove(bookmark.id))
     if (dupbmarks.length != 0) return
     const path = title.substring(0, title.lastIndexOf("/") + 1)
-    // TODO: if title is blank, get it from the page.
+
+
+    //this whole block grabs the current tab's title
+    //based on an async call since document.title won't have any meaning here
+    //
+    var currTitle;
+    function getCurrTabTitle(tabs) {
+        //function callback for promise for getting title of current tab
+        //requires [tabs] permission which tridactyl should have grabbed anyway
+        if (tabs[0]) {
+            var currentTab = tabs[0];
+            currTitle = currentTab.title
+        }
+    }
+    var titleSourceTab = browser.tabs.query({active: true, currentWindow: true});
+    //TODO: define and implement sane behaviour for if the title is blank, but the URL
+    //      that is being bookmarked is not actually the current tab. just using 
+    //      await activeTab().title doesn't make sense for that, we'll need to change the tab query
```

Comment:
You found `activeTab` after all? I guess your problem was the await - you need `(await activeTab()).title`, otherwise the dot tries to operate on the promise returned by activeTab. It would be much clearer if `await` was just a function rather than a magic operator but, you know, it's JavaScript...

Reply:
I only found that helper function in the last moments of development! 
I considered using activeTab after I found it, but wasn't sure about simplifying things in case I needed to edit the lookup again.